### DISCUSSION TITLE
NewLDAPIdentity

### DISCUSSION THREAD

**Discussion by macthev** [minio/minio#13301] [Q&A] [answered]
Testing out the sample program  https://github.com/minio/minio/blob/master/docs/sts/ldap.md to get STS creds. 

hitting issue with self signed certs on NewLDAPIdentity call.
x509: certificate signed by unknown authority

Is there a way to set insecure or can this be added to the transport opts like I was able to do with call?

TLSClientConfig: &tls.Config{InsecureSkipVerify: true},

minio.New(stsEndpointURL.Host, opts)

**Comment by harshavardhana** [comment, marked as answer]
> x509: certificate signed by unknown authority

Drop your public.crt in your client /usr/share/ssl or /etc/ssl - then run `update-ca-certificates` and also you can set custom transport with whatever root CAs you want to trust - clnt.SetCustomTransport(...)

**Comment by macthev** [comment]
SetCustomTransport seems not found in v7. 
only option appears in v6 
but not for github.com/minio/minio-go/v7/pkg/credentials -   cr.NewLDAPIdentity
The error is occurring during the call to cr.NewLDAPIdentity

This is what I would need set to allow insecure like I can od in the minio.New (...,opts) call.
Currently hard coded as this in its_ldap_identity.go
	l := LDAPIdentity{
		Client:       &http.Client{Transport: http.DefaultTransport},
The /etc/ssl -  is not working due self signed certs from the k8/routes generating wildcard cert and I can't request this *.myminio.svc target.
from with the pod I can use /etc/ssl.

**Reply by harshavardhana** [reply]
> SetCustomTransport seems not found in v7. 
> only option appears in v6 
> but not for github.com/minio/minio-go/v7/pkg/credentials -   cr.NewLDAPIdentity
> The error is occurring during the call to cr.NewLDAPIdentity
> 
> This is what I would need set to allow insecure like I can od in the minio.New (...,opts) call.
> Currently hard coded as this in its_ldap_identity.go
> 	l := LDAPIdentity{
> 		Client:       &http.Client{Transport: http.DefaultTransport},

You need to pass custom transport here.

> The /etc/ssl -  is not working due self signed certs from the k8/routes generating wildcard cert and I can't request this *.myminio.svc target.
> from with the pod I can use /etc/ssl. 
>
